platform/web/request.go

```diff
@@ -81,9 +101,9 @@ func validateRequestPayload(request *http.Request, model interface{}, options De
 	body, err := io.ReadAll(request.Body)
 	if err != nil {
 		if err.Error() == "http: request body too large" {
-			return fmt.Errorf("%w", NewErrRequestBodyTooLarge())
+			return fmt.Errorf("valid request payload failed: %w", ErrRequestBodyTooLarge)
 		}
-		return err
+		return fmt.Errorf("could not read request body: %v", err)
 	}
 	request.Body = io.NopCloser(bytes.NewBuffer(body))
 	payload := gojsonschema.NewBytesLoader(body)
```

Comment:
could we get rid of calls to `gojsonschema` if we have a `jsonschema` package ?

Reply:
I initially tried to do so, but it was getting messy to avoid breaking changes on the existing `DecodeWithJSONSchema` and `DecodeWithEmbeddedJSONSchema` functions. And since we wanted to deprecate them to just expose a JSON schema validation function and the decoding one, it was simpler to just not touch them.

Though I did forget to add the `Deprecated` comment on those 👍

Reply:
Actually I solved the problem that blocked me initially and I arrived to what seems like a good solution (using the new validate in the existing functions without deprecating them).

Hopefully it should be good now 😄